### PR TITLE
Add a secret GM notes field on items

### PR DESCRIPTION
**Pull request by nikolaj-a**
Add a GM notes field to the common item template, which is shown only to GM users in the description tab of all item sheets.

The specific use-case right now is for forged documents that need a secret DC to discover the forgery for passive observers, but the field itself should be usable in many situations.

image

### CODE REVIEW

**Comment by stwlam** [comment]
How big is it when it's empty?

**Comment by nikolaj-a** [comment]
Same size. I found it to be a little hard to adjust the size properly. Maybe it can be collapsed a bit using `:has()` or by converting the description tab to a grid?

I picked the current size by looking at this:
image

With the default item sheet window size, the editor buttons take up two rows making it a _super_ tight fit 😐

**Comment by nikolaj-a** [comment]
Another option is to add a GM tab. Not my preferred option due to the reduced visibility. But it does give room for more GM-y things.

**Comment by stwlam** [comment]
Maybe it can default to `null` and have minimal profile until the GM clicks +?

**Comment by nikolaj-a** [comment]
Let me try that.

**Comment by nikolaj-a** [comment]
For some reason the `:has(...)` selector doesn't work properly with descendant classes 🤷 This is the most compact I could make it, since the editor has to be present in the DOM for `Sheet#activeEditor(...)` to work.

image

**Comment by CarlosFdez** [comment]
You could check hazards to see how to make them compact. I had to do some shenaniganry there.

**Comment by nikolaj-a** [comment]
After taking a look at the hazard sheet, I decided to go in a slightly different direction and put the GM notes at the top and let the entire description panel flow normally, instead of having an always-visible, height-adapting block at the bottom.
image

File: src/module/item/sheet/base.ts [hunk 1]
```diff
@@ -379,6 +383,16 @@ export class ItemSheetPF2e<TItem extends ItemPF2e> extends ItemSheet<TItem> {
                 this.item.update({ [`system.variants.-=${index}`]: null });
             });
         }
+
+        for (const section of htmlQueryAll(html, "[data-action=edit-section]")) {
+            section.addEventListener("click", (event) => {
+                if (!(event.target instanceof HTMLElement)) return;
```
Comment (stwlam) [inline src/module/item/sheet/base.ts:389]: Is this not the same element as `section`?

File: static/templates/items/item-sheet.hbs [hunk 2]
```diff
@@ -82,11 +82,37 @@
...
             <!-- Item Description -->
             <div class="tab item-description" data-tab="description">
-                {{#if (not isVariant)}}
-                    {{editor enrichedContent.description target="system.description.value" button=true owner=owner editable=editable}}
-                {{else}}
-                    {{{enrichedContent.description}}}
-                {{/if}}
+                <div class="item-description-layout">
+                    {{#if user.isGM}}
+                        <div class="section">
```
Comment (stwlam) [inline static/templates/items/item-sheet.hbs:87]: A `div` with a class of "section" is rather perverse. :)

File: src/module/item/sheet/base.ts (hunk 1 above)
Comment (nikolaj-a) [reply src/module/item/sheet/base.ts:389]: It's _supposed_ to be the little 🗒️ icon in the header. I might be able to attach the listener to the section instead if that's better.

File: static/templates/items/item-sheet.hbs (hunk 2 above)
Comment (nikolaj-a) [reply static/templates/items/item-sheet.hbs:87]: Haha, good point! 😂 I'll change it to use `<section>` instead.

File: src/module/item/sheet/base.ts (hunk 1 above)
Comment (nikolaj-a) [reply src/module/item/sheet/base.ts]: Moved the listener to the container instead.

**Comment by shemetz** [comment]
Can you please change the "**GM Notes**" header to look different while it's empty?  I imagine that 99% of items will not have any GM notes, and for all of those cases the extra text and separating line take up a lot of space, along with drawing the eye.

I would prefer it if the GM Notes element didn't take any lines at all, as long as it's empty;  and the GM would need to click some button in order to add GM notes, with that button being unobtrusive and somewhere on the side or the top of the dialog window.

**Comment by nikolaj-a** [comment]
Compacted the description tab further when there are no GM notes:
Item sheet without GM notes

**Pull request merged**